pkg/certwatcher/certwatcher.go

```diff
@@ -93,7 +91,7 @@ func (cw *CertWatcher) Start(ctx context.Context) error {
 				files.Delete(f)
 			}
 			return true, nil
-		}); err != nil {
+		}); wait.Interrupted(err) {
```

Comment:
why this change is required?

Reply:
`err != nil` seems to be for `ErrWaitTimeout` only and the migration note suggests:

```
`Interrupted(err) bool` has been added that should be used instead of
checking `err == ErrWaitTimeout`. 
```

Reply:
> err != nil seems to be for ErrWaitTimeout

`err != nil` checks for all errors not only ErrWaitTimeout

> `Interrupted(err) bool` has been added that should be used instead of
checking `err == ErrWaitTimeout`. 

We didn't use `err == ErrWaitTimeout` before so not sure why we should use Interrupted now?

Reply:
I think it's fine to keep `err != nil` but what I wanted to say was the errors we are getting from the new func are different and the errors are covered by `wait.Interrupted()` since the condition never returns errors: "The returned error is the error returned by the last condition or the context error if the context was terminated."

Reply:
Sounds good!